core/src/utils/Log.cpp

```diff
@@ -0,0 +1,30 @@
+// Copyright (C) 2019-2020 Zilliz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under the License.
+#include "utils/Log.h"
+
+#include <cstdarg>
+#include <cstdio>
+#include <memory>
+#include <string>
+
+namespace milvus {
+
+std::string
+LogOut(const std::string& pattern, ...) {
+    auto str_p = std::make_unique<char[]>(pattern.length() + 256);
```

Comment:
what happen if message is larger than str_p allocated space size?

Reply:
May cause oom. It's a bug. So I need to replace `vsprintf` with `vsnprintf` to avoid this case.